src/main/java/uk/ac/ebi/atlas/hcalandingpage/HcaHumanExperimentDao.java

```diff
@@ -18,41 +22,46 @@
         /* Reason for checking blank here is in case characteristicValue is empty our solrQueryBuilder
          * builds query like characteristic_value:("\*") which is different from characteristic_value:*
          * which we need. So to handle that situation if characteristicValue is empty we build query
          * characteristic_name:("organism_part") as we need all the experiments in case of empty characteristicValue
          * */
-        queryBuilder
-                .setNormalize(false)
-                .addQueryFieldByTerm(CHARACTERISTIC_NAME, characteristicName)
+
+        queryBuilder.addQueryFieldByTerm(CHARACTERISTIC_NAME, characteristicName)
                 .setFieldList(EXPERIMENT_ACCESSION)
-                .setRows(10000000);
-        if (isNotBlank(characteristicValue)) {
-            queryBuilder
-                    .addQueryFieldByTerm(ImmutableMap.of(
-                            CHARACTERISTIC_VALUE, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION_ANCESTORS_LABELS, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION_ANCESTORS_URIS, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION_LABEL, ImmutableList.of("*".concat(characteristicValue))
-                    ));
+                .sortBy(EXPERIMENT_ACCESSION, SolrQuery.ORDER.asc)
+                .setRows(SolrQueryBuilder.DEFAULT_ROWS);
+
+        if(!characteristicValues.isEmpty()) {
+            queryBuilder.addQueryFieldByTerm(ImmutableMap.of(
+                    CHARACTERISTIC_VALUE, characteristicValues,
+                    ONTOLOGY_ANNOTATION, characteristicValues,
+                    ONTOLOGY_ANNOTATION_ANCESTORS_LABELS, characteristicValues,
+                    ONTOLOGY_ANNOTATION_ANCESTORS_URIS, characteristicValues,
+                    ONTOLOGY_ANNOTATION_LABEL, characteristicValues
+            ));
         }
 
-        var results = this.singleCellAnalyticsCollectionProxy.query(queryBuilder).getResults();
-        return results
-                .stream()
-                .map(solrDocument -> (String) solrDocument.getFieldValue(EXPERIMENT_ACCESSION.name()))
-                .collect(toImmutableSet());
+        var humanExperimentsSearchStream = new SearchStreamBuilder<>(singleCellAnalyticsCollectionProxy, queryBuilder);
+
+        var uniqueHumanExperimentsStream = new UniqueStreamBuilder(humanExperimentsSearchStream, EXPERIMENT_ACCESSION.name());
+
+        try(TupleStreamer tupleStreamer = TupleStreamer.of(uniqueHumanExperimentsStream.build())) {
+            return (tupleStreamer.get()
+                    .map(tuple -> (String) tuple.get(EXPERIMENT_ACCESSION.name()))
+                    .collect(toImmutableSet()));
+        }
```

Comment:
Why did we go from normal solr query to using solr stream expressions?

I believe in this case the performance with simple solr query is much faster than using stream expressions. For example when I ran test case `returnAllExperimentsForEmptyCharacteristicValue` with stream expression my laptop took `59 secs` to finish the test case whereas with normal solr query it just took `19 secs`.

Reply:
Why did we go from normal solr query to using solr stream expressions? - the approach suggested by Alfonso. We will check with him what he would suggest before changing.

Reply:
Regarding this, I guess once you have profiled what is the bottleneck in solr java code then we can decide which approach to take.